superset/viz.py

```diff
@@ -225,44 +224,46 @@ def get_json(self, force=False):
             self.get_payload(force),
             default=utils.json_int_dttm_ser, ignore_nan=True)
 
-    @property
-    def cache_key(self):
-        form_data = self.form_data.copy()
-        merge_extra_filters(form_data)
-        s = str([(k, form_data[k]) for k in sorted(form_data.keys())])
+    def cache_key(self, query_obj):
```

Comment:
We probably need to include the datasource itself => you can have 2 databases with the same tables, which as far as I can tell would cause collisions

Reply:
Good catch, I think so too.

Reply:
@fabianmenges/@mistercrunch I agree. I made the cache key a tuple of the database name and query string. PTAL.